File: Full_Stack_Capstone_Project/smartpizza_backend/src/main/java/com/smartpizza/service/CartService.java
package com.smartpizza.service;

import com.smartpizza.dto.request.AddToCartRequest;
import com.smartpizza.dto.request.UpdateCartItemRequest;
import com.smartpizza.dto.response.CartResponse;

public interface CartService {

    CartResponse getCart(String userEmail);

    CartResponse addToCart(String userEmail, AddToCartRequest request);

    CartResponse updateItem(String userEmail, UpdateCartItemRequest request);

    CartResponse removeItem(String userEmail, Long cartItemId);

    CartResponse clearCart(String userEmail);
}

File: Full_Stack_Capstone_Project/smartpizza_frontend/src/services/cartService.js
import apiClient from '../api/axiosClient';

/**
 * Cart API. The backend returns the full cart on every mutation:
 * { id, items: [...], totalItems, subtotal }.
 */
export const cartService = {
  getCart: () => apiClient.get('/cart').then((res) => res.data),

  addToCart: (payload) => apiClient.post('/cart/add', payload).then((res) => res.data),

  updateItem: (payload) => apiClient.put('/cart/update', payload).then((res) => res.data),

  removeItem: (cartItemId) =>
    apiClient.delete(`/cart/remove/${cartItemId}`).then((res) => res.data),

  clearCart: () => apiClient.delete('/cart/clear').then((res) => res.data),
};
